index.html

```diff
@@ -204,6 +204,70 @@ <h2>
+      <h2>
+        Model
+      </h2>
+      <p>
+        A <dfn data-lt="gamepad">gamepad</dfn> is a physical or virtual input
+        device that provides button and/or axis inputs. The
+        <dfn data-lt="host">host</dfn> is the operating system environment
+        hosting the user agent. The host MUST provide an algorithm for
+        <dfn data-lt="enumerating">enumerating</dfn> connected gamepads. A
+        gamepad is considered <dfn data-lt="connected">connected</dfn> when the
+        host is capable of reading new inputs. A gamepad that is not connected
+        is <dfn data-lt="disconnected">disconnected.
+      </p>
+      <p>
+        The host MUST provide an algorithm for synchronously
+        <dfn data-lt="reading input data">reading input data</dfn> from the
+        gamepad. If the host does not provide such an algorithm, the user agent
+        MAY implement an algorithm that caches the most recent input data and
+        synchronously returns the cached value.
+      </p>
+      <p>
+        When a gamepad transitions from disconnected to connected, it
+        <dfn data-lt="becomes connected">becomes connected</dfn>. Likewise,
+        when a gamepad transitions from connected to disconnected, it
+        <dfn data-lt="becomes disconnected">becomes disconnected</dfn>. The
+        host MUST provide algorithms to register listeners for these
+        transitions. If these events are not available, the user agent MAY
+        implement an algorithm that synthesizes the events by periodically
+        enumerating connected gamepads.
+      </p>
+      <p>
+        The user agent MUST define a
+        <dfn data-lt="gamepad user gesture">gamepad user gesture</dfn>. A
+        gamepad user gesture is a class of gamepad input state transitions that
+        signify that the user is interacting with the gamepad. At a minimum, the
+        gamepad user gesture MUST include state changes where any button
+        transitions from unpressed to pressed. Other transitions MAY be included
+        as user gestures. A transition MUST NOT be included if the transition
+        could occur without user interaction.
+      </p>
+      <p>
+        The user agent MUST NOT expose information about connected gamepads
+        until a gamepad user gesture has been received from any connected
+        gamepad. The user gesture MUST be tracked independently for each gamepad
+        context. A <dfn data-lt="gamepad context">gamepad context</dfn> MUST be
+        created when a page first requests the current gamepad state or
+        registers for gamepad connection events. The gamepad context MUST be
+        destroyed when the tab is closed or navigates away from the current
+        page. Reloading the current page MUST NOT destroy the gamepad context.
+      </p>
+      <p>
+        To register a user gesture, a gamepad context MUST be active. A gamepad
+        context is active if the page associated with the context is visible and
```

Comment:
```suggestion
        context is <dfn>active</dfn> if the page associated with the context is visible and
```

Reply:
Do I need to add a reference to the spec somewhere else? These are getting red squiggly underlines.

Reply:
This is blocked on getting Visibility API republished... filed a bug. https://github.com/w3c/page-visibility/issues/65